m4/compiler.m4

```diff
@@ -13,18 +13,15 @@ AC_DEFUN([LIBMESH_SET_COMPILERS],
      AC_ARG_ENABLE(mpi,
                    AS_HELP_STRING([--disable-mpi],
                                   [build without MPI message passing support]),
-                   [case "${enableval}" in
-                     yes)  enablempi=yes ;;
-                     no)  enablempi=no ;;
-                     *)  AC_MSG_ERROR(bad value ${enableval} for --enable-mpi) ;;
-                   esac],
+                   [AS_CASE("${enableval}",
+                            [yes], [enablempi=yes],
+                            [no],  [enablempi=no],
+                            [AC_MSG_ERROR(bad value ${enableval} for --enable-mpi)])],
                    [enablempi=yes])
 
-  if  (test "$enablempi" != no) ; then
-    CXX_TRY_LIST="mpicxx mpiCC mpicc $CXX_TRY_LIST"
-  else
-    AC_MSG_RESULT(>>> Disabling MPI per user request <<<)
-  fi
+  AS_IF([test "$enablempi" != no],
+        [CXX_TRY_LIST="mpicxx mpiCC mpicc $CXX_TRY_LIST"],
+        AC_MSG_RESULT([>>> Disabling MPI per user request <<<]))
```

Comment:
Any particular reason we're wrapping this AC_MSG_RESULT argument and not the similar others below?

Reply:
Oh yeah... I was playing around with different amounts of quoting (inside the argument to `AC_MSG_RESULT`, around the entire `AC_MSG_RESULT` call, etc.) and in the end I don't think it made any difference. We could change this one to be consistent with all the others if you prefer...

Reply:
Nah, it's fine; I just wanted to make sure there wasn't some subtle m4 gotcha I was missing.